kythe/go/platform/tools/kzip/mergecmd/mergecmd.go

```diff
@@ -165,6 +168,9 @@ func mergeInto(ctx context.Context, wr *kzip.Writer, path string, filesAdded str
 					return fmt.Errorf("error closing file: %v", err)
 				}
 			}
+			if vname, match := c.rules.Apply(ri.Info.Path); match {
+				proto.Merge(ri.VName, vname)
```

Comment:
I'm curious if instead this should be assignment, rather than merge?

Reply:
IMO, no.  I think it makes more sense to leave the unspecified fields alone rather than clear them.

Reply:
(Which, in this case, would be "corpus")

Reply:
I don't think this should be a merge.  At least `corpus` and `root` should be treated as a unit.

Reply:
Due to the use of proto3, this actually has to be assignment or there is no way to set an empty root :-(